core/domain/skill_services.py

```diff
@@ -192,6 +192,23 @@ def get_multi_skill_summaries(skill_ids):
     return skill_summaries
 
 
+def get_multi_skills(skill_ids):
+    """Returns a list of skills matching the skill IDs provided.
+
+    Args:
+        skill_ids: list(str). List of skill IDs to get skills for.
+
+    Returns:
+        list(Skill). The list of skills matching the provided IDs.
```

Comment:
This does not seem to reflect the implementation just excluding skills that are None.

Either throw an exception if a skill is requested that does not exist, or return None in the appropriate position; probably the former is better if you don't expect this to occur in practice. But just silently removing stuff altogether seems like it's not the right thing to do -- you'll end up with the length of the input list not matching the output list which is something the caller won't expect given the function contract.

There should also be tests for this error case, it's quite important.

Reply:
Fixed by raising Exception